graphql/queryResolvers/moduleCrud.ts

```diff
@@ -0,0 +1,29 @@
+import type { MutationDeleteModuleArgs, MutationAddModuleArgs } from '..'
+import prisma from '../../prisma'
+
+export const modules = async () => {
+  return prisma.module.findMany({
```

Comment:
Hmm `findMany` should already return a promise, right? If that's the case, then it is redundant to make this function async

Reply:
Actually, I think it'd be better to keep the function async, and then wrap the promise in a try/catch block
```
try {
  const allModules = await prisma.module.findMany(...)
} catch(err) {
  // SOME LOGGING
  // error handling logic
}
```

Reply:
@anthonykhoa good point about `async.

also your sample code is wrong, `try/catch` does not work for promises. You would have to do:

```js
return prisma.module.findMany(...).catch(...)
```

We could return empty array on errors so we always guarantee returning an array:

```js
return prisma.module.findMany(...).catch( _e => return [])
```

Reply:
Ah good idea. returning empty array is appropriate for this scenario.